paasta_tools/utils.py

```diff
@@ -677,20 +678,22 @@ class NoConfigurationForServiceError(Exception):
     pass
 
 
-def get_readable_files_in_glob(input_glob):
+def get_readable_files_in_glob(glob, path):
     """
-    Returns lexicographically-sorted list of files that are readable in an input glob
+    Returns a sorted list of files that are readable in an input glob by recusivly searching a path
     """
-    files = []
-    for f in sorted(glob.glob(input_glob)):
-        if os.path.isfile(f) and os.access(f, os.R_OK):
-            files.append(f)
-    return files
+    globbed_files = []
+    for root, dirs, files in os.walk('/etc/paasta'):
+        for f in files:
+            fn = os.path.join(root, f)
+            if os.path.isfile(fn) and os.access(fn, os.R_OK) and fnmatch(fn, glob):
+                globbed_files.append(fn)
+    return sorted(globbed_files)
 
 
 def load_system_paasta_config(path=PATH_TO_SYSTEM_PAASTA_CONFIG_DIR):
     """
-    Reads Paasta configs in specified directory in lexicographical order and merges duplicated keys (last file wins)
```

Comment:
Well, if we're going to say which file wins (`(last file wins)`), we should specify what we mean by "last".